OloEngine/src/OloEngine/Serialization/MeshBinarySerializer.cpp

```
@@ -0,0 +1,1944 @@
+    // ========================================================================
+    // Internal helpers
+    // ========================================================================
+
+    namespace
+    {
+        // ── Validation helpers ──
+
+        // Returns true if every component in the matrix array is finite.
+        // On failure, logs the first offending element and replaces the matrix
+        // with identity so downstream code doesn't propagate NaN/Inf.
+        bool ValidateMat4Array(std::vector<glm::mat4>& arr, const char* arrayName,
```

Clang (14.0.6)

[warning] variable 'ValidateMat4Array' is non-const and globally accessible, consider making it const

(cppcoreguidelines-avoid-non-const-global-variables)